File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        map<char, char> dict1;
        map<char, char> dict2;
        for (int i = 0; i < s.size(); i++){
            if(dict1.find(s[i]) != dict1.end()){ 
                if(dict1[s[i]] != t[i]){
                    return false;
                }
            }
            else {
                dict1[s[i]] = t[i];
            }
            if(dict2.find(t[i]) != dict2.end()){ 
                if(dict2[t[i]] != s[i]){
                    return false;
                }
            }
            else {
                dict2[t[i]] = s[i];
            }
        }
        return true; 
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        s2t = dict()
        t2s = dict()
        for i in range(len(s)):
            if s[i] not in s2t:
                if t[i] not in t2s:
                    s2t[s[i]] = t[i]
                    t2s[t[i]] = s[i]
                else:
                    return False
            elif s2t[s[i]] != t[i]:
                return False
        
        return True